src/components/web-form-renderer.vue

```diff
@@ -0,0 +1,336 @@
+  let url = apiPaths.submissionAttachment(form.projectId, form.xmlFormId, false, primaryInstance.instanceId, attachmentDescriptor.file.name);
+  if (form.draftToken) {
+    url = url.replace(/^\/v1\//, `/v1/test/${form.draftToken}/`);
+  }
+  if (route.query.st) {
+    url += `?st=${route.query.st}`;
+  }
+  await request({
+    method: 'POST',
+    url,
+    data: attachmentDescriptor.file,
+    onUploadProgress: (progressEvent) => {
+      attachmentDescriptor.progress = Math.floor((progressEvent.loaded * 100) / progressEvent.total);
+    },
+  })
+    .then(() => {
+      attachmentDescriptor.uploadSuccess = true;
+    })
+    .catch(() => {
+      Object.assign(attachmentDescriptor, { uploadSuccess: false, progress: null });
+    });
+};
+
+const postSubmission = async () => {
+  // TODO: add logic for edit submission
+
+  if (await postPrimaryInstance()) {
+    // eslint-disable-next-line no-unused-vars
+    for (const [ix, _] of attachmentUploads.entries()) {
+      // Post each attachment in turn, and not in parallel. Parallel transfers means more time spent per transfer, which
+      // under adverse network conditions increases the probability that the transfer does not complete. And since uploads
+      // as of yet are not resumable, this would mean restarting the upload from 0 again.
+      // eslint-disable-next-line no-await-in-loop
+      await uploadAttachment(ix);
+    }
+
+    return true;
+  }
+
+  return false;
+};
+
+
+/**
+ * When WebForms's submit button is clicked, it dispatches an event which is handed to
+ * this handler, which can then upload the form and its attachments as present in the
+ * event payload.
+ */
+const handleSubmit = async (payload) => {
+  // TODO: waiting for web-forms v0.7, current version doesn't return any payload.
+  // eslint-disable-next-line no-constant-condition
+  if (props.actionType === 'preview') {
+    showModal('preview');
+  } else {
+    // eslint-disable-next-line no-unused-vars
+    const { data, definition, status, violations } = payload;
+    if (status !== 'ready') {
+      // Status is not ready when Form is not valid and in that case submit button will be disabled,
+      // hence this branch should never execute.
+      return;
+    }
+
+    primaryInstance.file = data.instanceFile;
+    data.attachments.forEach((file, ix) => { attachmentUploads[ix] = { file, progress: null, uploadSuccess: null }; });
+
+    const result = await postSubmission();
+
+    if (result) {
+      if (isPublicLink.value) {
+        showModal('thankYou');
+        formVersionXml.reset();
+      } else {
+        showModal('submission');
+      }
+    }
+
+    // TODO: redirect or clear Form or say do nothing based on the workflow new/edit/public-link
+    // related issue getodk/central#928
+  }
+};
+
+</script>
+
+<style lang="scss">
+@import '../assets/scss/_variables.scss';
+
+:root {
+  font-size: 16px;
+}
+html, body {
+  background-color: var(--gray-200);
+  box-shadow: none;
+}
```

Comment:
Is it possible for the user to navigate from a page that renders `WebFormRenderer` to a page that doesn't render `WebFormRenderer`? If so, I think we need to do something so that these styles aren't applied on the page that doesn't render the `WebFormRenderer`. Something like:

```scss
html, body {
  &:has(.odk-form) {
    background-color: var(--gray-200);
    box-shadow: none;
  }
}
```